### ISSUE TITLE
Switch to `kyaml` for manifest parsing

### ISSUE THREAD

**Issue by marshall007** [sigstore/cosign#1263]
The current implementation of `cosign manifest verify` is not extensible and, for example, cannot detect image references in CRDs (like the steps in a Tekton `Task`):

https://github.com/sigstore/cosign/blob/8194edd77d1077505e85e89122432479a63791e0/cmd/cosign/cli/manifest/verify.go#L102-L145

We could make this more robust and also drop a dependency on `k8s.io/apimachinery` by switching to `sigs.k8s.io/kustomize/kyaml`. See the following example of what this would look like:

https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml/yaml#example-Lookup-Sequence

**Comment by marshall007** [comment]
Might also make sense for these commands to support reading from `stdin`. That way you can pipe directly from things like `kubectl get`, `helm template`, etc.

**Comment by dlorenc** [comment]
cc @mattmoor I think this makes sense!

**Comment by developer-guy** [comment]
I can take care of that one if you folks let me @marshall007 @dlorenc

**Comment by dlorenc** [comment]
Go for it!

**Comment by erkanzileli** [comment]
When we inspected this example, we see that we need the path specifically. Did you may be asked for a flag to specify the image path in the command for specific CRD.

**Example:**

```yaml
apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: example-task-name
spec:
  steps:
    - name: ubuntu-example
      image: ubuntu
```

Here is how to verify this specific CRD

```shell
$ cosign manifest verify --path=spec.steps.*.image tekton-task.yaml
```

WDYT @marshall007 @dlorenc

**Comment by marshall007** [comment]
> When we inspected this example, we see that we need the path specifically. Did you may be asked for a flag to specify the image path in the command for specific CRD.

@erkanzileli that's right, I was thinking you'd be able to append additional/custom paths via a CLI flag. I think you should be able to specify this flag multiple times as you might need to add multiple field paths. I also think this should _append_ to the default set as I see no reason why you would ever not want to include the default paths we look at currently.

@developer-guy you can definitely take this one. I was originally planning to but I'm pretty busy so you will almost certainly beat me to it!
